spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePublisherCallbacksIntegrationTests.java

```diff
@@ -205,26 +190,18 @@ public Void doInRabbit(Channel channel) throws Exception {
 	public void testPublisherConfirmWithSendAndReceive() throws Exception {
 		final CountDownLatch latch = new CountDownLatch(1);
 		final AtomicReference<CorrelationData> confirmCD = new AtomicReference<CorrelationData>();
-		templateWithConfirmsEnabled.setConfirmCallback(new ConfirmCallback() {
-
-			@Override
-			public void confirm(CorrelationData correlationData, boolean ack, String cause) {
-				confirmCD.set(correlationData);
-				latch.countDown();
-			}
+		templateWithConfirmsEnabled.setConfirmCallback((correlationData, ack, cause) -> {
+			confirmCD.set(correlationData);
+			latch.countDown();
 		});
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(this.connectionFactoryWithConfirmsEnabled);
 		container.setQueueNames(ROUTE);
-		container.setMessageListener(new MessageListenerAdapter(new Object() {
-
-			@SuppressWarnings("unused")
-			public String handleMessage(String in) {
-				return in.toUpperCase();
-			}
-		}));
+		container.setMessageListener(
+				new MessageListenerAdapter((ReplyingMessageListener<String, String>) in -> in.toUpperCase()));
```

Comment:
>And also I think this Lambda deserves some Docs in favor of Java 8 universe 

Not sure what you mean. Do you mean convert any example code in the docs to lambdas?